compiler/semexprs.nim

```diff
@@ -227,7 +229,7 @@ proc semCast(c: PContext, n: PNode): PNode =
   if tfHasMeta in targetType.flags:
     localError(n.sons[0].info, errCastToANonConcreteType, $targetType)
   if not isCastable(targetType, castedExpr.typ):
-    localError(n.info, errExprCannotBeCastToX, $targetType)
+    localError(n.info, errExprCannotBeCastToX, $targetType & "=" & typeToString( targetType, preferDesc))
```

Comment:
This can produce weird messages like 'proc () = proc ()' so you need to add more logic here:

let tar = $targetType; let alt = typeToString(targetType, preferDesc);  if tar != alt:  ...